pkg/remote/azurerm/repository/privatedns.go

```diff
@@ -145,6 +146,30 @@ func (s *privateDNSRepository) ListAllAAAARecords(zone *armprivatedns.PrivateZon
 	return results, nil
 }
 
+func (s *privateDNSRepository) ListAllCNAMERecords(zone *armprivatedns.PrivateZone) ([]*armprivatedns.RecordSet, error) {
+	cacheKey := fmt.Sprintf("privateDNSListAllCNAMERecords-%s", *zone.ID)
+	if v := s.cache.Get(cacheKey); v != nil {
+		return v.([]*armprivatedns.RecordSet), nil
+	}
+
+	records, err := s.listAllRecords(zone)
+	if err != nil {
+		return nil, err
+	}
+	results := make([]*armprivatedns.RecordSet, 0)
+	for _, record := range records {
+		if record.Properties.CnameRecord == nil {
+			continue
+		}
```

Comment:
This is not a efficient usage of the cache, we are calling the `listAllRecords` one time per record type with this approach.

I think this logic of record type check should be moved in the provider. In the repository we should have only a method that return all the records for a zone IMHO, so we can cache all results.

Reply:
I removed cache for the filtering and leaved it only for the request for all record.
I like having the ListAll<Type> in the repository as we could as well do filtering in the request and so this make the abstraction more transparent.

Reply:
Yeah sorry, seeing caching logic here put me on the wrong direction 🙏🏻 Thanks to have removed it, it does not add a real performance improvement since it may be most costly to acquire a lock on the cache map than filtering at some point, thanks to have removed it 👍🏻 

> I like having the ListAll{Type} in the repository as we could as well do filtering in the request and so this make the abstraction more transparent.

I agree with that 👍🏻